src/components/search/FacetListBase.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { useCallback, useContext } from 'react';
+import PropTypes from 'prop-types';
+
+import { NO_OPTIONS_FOUND } from './data/constants';
+import FacetDropdown from './FacetDropdown';
+import FacetItem from './FacetItem';
+import { SearchContext } from './SearchContext';
+import {
+  addToRefinementArray, setRefinementAction, deleteRefinementAction, removeFromRefinementArray,
+} from './data/actions';
+
+const FacetListBase = ({
+  attribute,
+  facetValueType,
+  isBold,
+  isCheckedOrRefinedField,
+  items,
+  title,
+}) => {
+  /**
+   * Handles when a facet option is toggled by either updating the appropriate
+   * query parameter for the facet attribute, or removes the facet attribute if
+   * there's no longer any selected options for that facet attribute.
+   */
+
+  const { refinementsFromQueryParams, dispatch } = useContext(SearchContext);
+
+  const handleInputOnChange = (item) => {
+    if (item.value && facetValueType === 'array') {
+      if (item.value.length > 0) {
+        if (refinementsFromQueryParams[attribute]?.includes(item.label)) {
+          dispatch(removeFromRefinementArray(attribute, item.label));
+        } else {
+          dispatch(addToRefinementArray(attribute, item.label));
+        }
+      } else {
+        dispatch(deleteRefinementAction(attribute));
+      }
+    } else if (facetValueType === 'bool') {
+      // eslint-disable-next-line no-bitwise
+      dispatch(setRefinementAction(attribute, refinementsFromQueryParams[attribute] ^ 1));
+    }
+  };
+
+  const renderItems = useCallback(
+    () => {
+      if (!items || !items.length) {
+        return <span className="py-2 px-2">{NO_OPTIONS_FOUND}</span>;
+      }
+
+      return items.map(item => {
+        const isCheckOrRefined = isCheckedOrRefinedField ? item[isCheckedOrRefinedField] : !!item.value;
+
+        return (
+          <FacetItem
+            key={item.label}
+            handleInputOnChange={handleInputOnChange}
+            item={item}
+            isChecked={isCheckOrRefined}
+            isRefined={isCheckOrRefined}
```

Comment:
Do you mind reminding me what the difference between `isChecked` and `isRefined` is again? Drawing a blank 😃

Reply:
One controls whether the item is checked, and the other controls whether or not it's bolded. (The `is-refined` css class). 
At one point it made sense for them to be separate things, I think it might not at this point. I can consolidate if you think that makes sense.

Reply:
Consolidated!